### PR TITLE
Misc fixes (including tri)

### LINKED ISSUE #9654: Labels not saved in serialized MultiLayerNetwork
#### Issue Description
When saving a `MultiLayerNetwork` using the `ModelSerializer` the `labels` field is not persisted when loaded back into memory. This requires additional meta-data to be kept with the trained models. This might not be evident for those cases where classes are standardize into one-hot-encoding and always have a one-to-one mapping from column to class "number". But consider a generic case where classes could be e.g. [-1, +1], that mapping can be conveniently saved with `model.setLabels(INDArray)` and the output from e.g. `model.predict(INDArray)` can easily be translated back into the original labels found in the training data using the output from the `model.getLabels()` method.

I might be missing the true reason for the field `labels`, if this use case is not the reason for having it. There is no javadoc for either the field, setter or getter methods, so it's very hard to tell what its purpose is.

#### Version Information
Tested both in 1.0.0-M1.1 and 1.0.0-M2 on Linux using nd4j-native and nd4j-native-platform backends.

### LINKED ISSUE #9655: Zero SameDiff Gradients
#### Issue Description

I've written a minimal code to reproduce the issue:
```
        SameDiff sd = SameDiff.create();

        SDVariable i0 = sd.placeHolder("i0", DataType.FLOAT, 2,5);
        SDVariable w0 = sd.var("w0", new OneInitScheme('c'), DataType.FLOAT, 5, 3);
        SDVariable b0 = sd.var("b0", new OneInitScheme('c'), DataType.FLOAT,3);

        SDVariable w1 = sd.var("w1", new OneInitScheme('c'), DataType.FLOAT, 3,3);
        SDVariable b1 = sd.var("b1", new OneInitScheme('c'), DataType.FLOAT,3);

        SDVariable i1 = i0.mmul(w0).add(b0);
        SDVariable i2 = i1.mmul(w1).add(b1).add(i1);
        SDVariable l = i2.sum();

        sd.setLossVariables(l);
        INDArray gd = sd.calculateGradients(Map.of("i0",Nd4j.rand(2,5)),"w0").get("w0");
        System.out.println(gd);
```

I would expect non-zero gradients, instead, I get:

```
        [[         0,         0,         0], 
         [         0,         0,         0], 
         [         0,         0,         0], 
         [         0,         0,         0], 
         [         0,         0,         0]]
```

#### Version Information
I am using the following packages:

```
nd4j-native-platform, version: 1.0.0-M2
deeplearning4j-datasets, version: 1.0.0-M2
deeplearning4j-core, version: 1.0.0-M2
```

#### Additional Information
The issue does not appear when using:
```
nd4j-native-platform, version: 1.0.0-M1.1
deeplearning4j-datasets, version: 1.0.0-M1.1
deeplearning4j-core, version: 1.0.0-M1.1
```

### LINKED ISSUE #9668: OmpNumThreads is hardcoded to 4 in M2 Release
#### Issue Description

- expected behavior - NativeOpsHolder allows to set the ompNumThreads amount based on OMP_NUM_THREADS environment variable
- encountered behavior - In org.nd4j.nativeblas.NativeOpsHolder#NativeOpsHolder(), line 107 this variable is hardcoded to 4

#### Version Information

* Deeplearning4j version - M2
* Platform information (OS, etc) - Windows 10
* CUDA version, if used - N/A
* NVIDIA driver version, if in use - N/A

#### Additional Information

N/A

#### Contributing

The fix is simply removing the mentioned above line 107. The correct assignment is already done in lines 94-106

### PR DESCRIPTION
**Pull request by agibsonccc**
## What changes were proposed in this pull request?
1. Fixes tri related bugs
2. Fix #9655 
3. Fix #9654 
4. Fix https://github.com/eclipse/deeplearning4j/issues/9668
(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

## Quick checklist

The following checklist helps ensure your PR is complete:

- [ ] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.org/eclipse-contributors) page for details
- [ ] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [ ] Created tests for any significant new code additions.
- [ ] Relevant tests for your changes are passing.

### CODE REVIEW

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/internal/memory/ArrayCacheMemoryMgr.java [hunk 1]
```diff
@@ -58,7 +56,7 @@ public ArrayCacheMemoryMgr() {
     }
 
     /**
-     * @param maxMemFrac             Maximum memory fraction to use as cache
+     * @param maxMemFrac             Maximum memory fraciton to use as cache
```
Comment (treo) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/internal/memory/ArrayCacheMemoryMgr.java]: typo

File: platform-tests/src/test/kotlin/org/eclipse/deeplearning4j/frameworkimport/frameworkimport/onnx/importer/TestOnnxFrameworkImporter.kt [hunk 2]
```diff
@@ -61,6 +61,7 @@ class TestOnnxFrameworkImporter {
         trainingConfig.minimize("loss")
         result.trainingConfig = trainingConfig.build()
 
+        println(result.summary())
```
Comment (treo) [inline platform-tests/src/test/kotlin/org/eclipse/deeplearning4j/frameworkimport/frameworkimport/onnx/importer/TestOnnxFrameworkImporter.kt]: left over from debugging?

**Review by treo — CHANGES_REQUESTED** [review]

**Review by treo — APPROVED** [review]
